src/handler.ts

```diff
@@ -0,0 +1,195 @@
+
+    const response = await respondWithUser(userId, container);
+    return response;
+  } catch (e) {
+    return {
+      statusCode: 500,
+      body: JSON.stringify({ error: e.message })
+    };
+  }
+};
+
+export const createUser: APIGatewayProxyHandler = async event => {
+  try {
+    assert(event.body, "Invalid request");
+    const { name, dob, address, description } = JSON.parse(
+      event.body as string
+    );
+
+    // ensure all fields are truthy
+    assert(name && dob && address && description, "Invalid body");
+
+    const container = retrieveContainer();
+    const { client, TableName } = container;
+    const id = uuid();
+    const now = Date.now();
+
+    const dynamoParams = {
+      Item: {
+        id,
+        name,
+        dob,
+        address,
+        description,
+        createdAt: now,
+        updatedAt: now
+      },
+      TableName
+    };
+
+    await client.put(dynamoParams).promise();
+
+    const response = await respondWithUser(id, container);
+    return response;
+  } catch (e) {
+    return {
+      statusCode: 500,
+      body: JSON.stringify({ error: e.message })
+    };
+  }
+};
+
+export const updateUser: APIGatewayProxyHandler = async event => {
+  try {
+    const userId = event.pathParameters!.userId;
+    assert(userId && event.body, "Invalid request");
+    const container = retrieveContainer();
+    const { TableName, client } = container;
+    const body = JSON.parse(event.body as string);
+    const updateFields = ["name", "dob", "address", "description"].filter(
+      field => body[field]
+    );
+
+    // ensure we have at least one field
+    assert(updateFields.length, "Invalid Body");
+
+    const now = Date.now();
+    const updateExpressionList = [...updateFields, "updatedAt"].map(
+      field => `#${field} = :${field}`
+    );
+
+    const ExpressionAttributeNames = updateFields.reduce(
+      (memo, field) => ({
+        ...memo,
+        [`#${field}`]: field
+      }),
+      { ["#updatedAt"]: "updatedAt" }
+    );
+
+    const ExpressionAttributeValues = updateFields.reduce(
+      (memo, field) => ({
+        ...memo,
+        [`:${field}`]: body[field]
+      }),
+      { [":updatedAt"]: now }
+    );
+
+    const dynamoParams = {
+      TableName,
+      Key: { id: userId },
+      UpdateExpression: `SET ${updateExpressionList.join(", ")}`,
+      ExpressionAttributeNames,
+      ExpressionAttributeValues
+    };
+
+    await client.update(dynamoParams).promise();
+
+    const response = await respondWithUser(userId, container);
+    return response;
+  } catch (e) {
+    return {
+      statusCode: 500,
+      body: JSON.stringify({ error: e.message })
+    };
+  }
+};
+
+export const deleteUser: APIGatewayProxyHandler = async event => {
+  try {
+    const userId = event.pathParameters!.userId;
+    assert(userId, "Invalid request");
+    const container = retrieveContainer();
+
+    const user = await getUser(userId, container);
+    assert(user, "User Not Found");
```

Comment:
Now I must go see what Roy Fielding said about this.

It could definitely be considered a success depending on context, but if I would lean toward sending a more informative status code -- perhaps a 404, so that the client can make an informed decision based on the response.